source/request-response/request_response_client.c

```diff
@@ -796,6 +1130,145 @@ static uint64_t s_mqtt_request_response_client_get_next_service_time(struct aws_
     return UINT64_MAX;
 }
 
+static struct aws_byte_cursor s_aws_mqtt_rr_operation_get_subscription_topic_filter(
+    struct aws_mqtt_rr_client_operation *operation) {
+    if (operation->type == AWS_MRROT_REQUEST) {
+        return operation->storage.request_storage.options.subscription_topic_filter;
+    } else {
+        return operation->storage.streaming_storage.options.topic_filter;
+    }
+}
+
+/* TODO: add aws-c-common API? */
+static bool s_is_operation_in_list(const struct aws_mqtt_rr_client_operation *operation) {
```

Comment:
Trivial: If this is added to aws-c-common it'll have a description that clarifies what it's doing but just looking at the name of the function it's not immediately clear what it's doing. Or maybe it is since its only argument is an operation and not a list to check whether the operation is in that list or not. Either way, never thought about a check for whether a node is part of a list or hasn't been added to one yet.

Reply:
Well it would be is_node_in_list which feels somewhat self-explanatory

Reply:
https://github.com/awslabs/aws-c-common/pull/1103